internal/notif/ably_notif.go

```diff
@@ -24,27 +26,57 @@ type PushNotificationBody struct {
 	Body  string `json:"body"`
 }
 
-type PushNotificationRecipient struct {
+type ApnsPushNotificationRecipient struct {
 	TransportType transportType `json:"transportType"`
 	DeviceToken   string        `json:"deviceToken"`
 }
 
+type FcmPushNotificationRecipient struct {
+	TransportType     transportType `json:"transportType"`
+	RegistrationToken string        `json:"registrationToken"`
+}
+
 type ApplePushNotification struct {
-	Recipient    PushNotificationRecipient `json:"recipient"`
-	Notification PushNotificationBody      `json:"notification"`
+	Recipient    ApnsPushNotificationRecipient `json:"recipient"`
+	Notification PushNotificationBody          `json:"notification"`
+}
+
+type AndroidPushNotification struct {
+	Recipient    FcmPushNotificationRecipient `json:"recipient"`
+	Notification PushNotificationBody         `json:"notification"`
 }
 
 func SendPushNotification(ctx context.Context, config config.AblyConfig, device *model.UserDevice, pushBody PushNotificationBody) (bool, error) {
 	if !config.Enabled {
 		return true, nil
 	}
-	body, err := json.Marshal(ApplePushNotification{
-		Recipient: PushNotificationRecipient{
-			TransportType: "apns",
-			DeviceToken:   *device.Token,
-		},
-		Notification: pushBody,
-	})
+
+	if device == nil || device.Token == nil {
+		return false, errors.New("Device token must not be nil")
+	}
+
+	var body []byte = nil
+	var err error = nil
+	if strings.ToLower(device.Platform) == "android" {
+		body, err = json.Marshal(AndroidPushNotification{
+			Recipient: FcmPushNotificationRecipient{
+				TransportType:     "fcm",
+				RegistrationToken: *device.Token,
+			},
+			Notification: pushBody,
+		})
+	} else if strings.ToLower(device.Platform) == "ios" {
+		body, err = json.Marshal(ApplePushNotification{
+			Recipient: ApnsPushNotificationRecipient{
+				TransportType: "apns",
+				DeviceToken:   *device.Token,
+			},
+			Notification: pushBody,
+		})
+	} else {
+		return false, errors.New("Unknown platform for device: " + device.Platform)
```

Comment:
nit: Generally it's preferable to use `fmt.Errorf("blah blah %s", device.Platform)

Reply:
Ok. In this case, do we still return the error tho?